prometheus_client/context_managers.py

```diff
@@ -27,11 +26,11 @@ def __exit__(self, typ: Optional[Type[BaseException]], value: Optional[BaseExcep
         return False
 
     def __call__(self, f: "F") -> "F":
-        def wrapped(func, *args, **kwargs):
+        @functools.wraps(f)
+        def wrapped(*args: Any, **kwargs: Any) -> Any:
             with self:
-                return func(*args, **kwargs)
-
-        return decorate(f, wrapped)
+                return f(*args, **kwargs)
+        return wrapped  # type: ignore
```

Comment:
It looks like there are two options for now: ignore or cast. 

PS In python 3.10 it was resolved via https://docs.python.org/3/library/typing.html#typing.ParamSpec.